estimagic/optimization/check_constraints.py

```diff
@@ -0,0 +1,123 @@
+import warnings
+
+import pandas as pd
+
+
+def check_compatibility_of_constraints(constraints, params, fixed):
+    """Compatibility checks for constraints.
+
+    Args:
+        constraint (dict)
```

Comment:
constraints and add explanation.

Reply:
Thanks. I also added docstrings to the remaining functions in that module.